Fix `Assert-AzureServicePrincipalForRbac`

JamesDawson opened the pull request:
Resolves an issue where this cmdlet was not handling the more recent property names used to hold the credential details.

Originally they were `appId`, `password` & `tenant`; this has since changed to `clientId`, `clientSecret`, `tenantId`.

The cmdlet will now detect when an existing KV secret is using the old format and will force a secret rotation in order to update the structure.

Also adds some test coverage for this new logic.

JamesDawson committed:
Add handling to support that the property name which holds the secret has changed over time - originally `password`, now `clientSecret`.

JamesDawson committed:
Updated related Pester tests

MikeEvansLarah commented on module/functions/azure/aad/Assert-AzureServicePrincipalForRbac.ps1:
```diff
@@ -1,314 +1,326 @@
+            Write-Host "Storing client secret in key vault [VaultName=$KeyVaultName, SecretName=$KeyVaultSecretName]"
+            Set-AzKeyVaultSecret -VaultName $KeyVaultName `
+                                 -Name $KeyVaultSecretName `
+                                 -SecretValue ($appLoginDetails | ConvertTo-Json | ConvertTo-SecureString -AsPlainText -Force) `
```
Should the property in `$appLoginDetails` be updated to `clientSecret`? I'm a bit confused as to where the `clientSecret` property is coming from if not from heer?

https://github.com/corvus-dotnet/Corvus.Deployment/pull/109/files#diff-92b916981b3830e7e32d533fcf5335a5b9c0e19872cd130333826c303b93e590R217

JamesDawson replied:
Yes, I wasn't quite sure how far to go with it.  The original source of this 'chosen' format stems from what used to be required by the `azure/login` GitHub Action - which matched what the Azure CLI produces when you create a service principal via `az ad sp create-for-rbac`.

Now the GH Action [documents](https://github.com/azure/login?tab=readme-ov-file#creds) the usage of `clientSecret` instead, but clearly still supports the old property too (otherwise our usage would be broken in a different way).

The intent for this change was to at least be compatible with both formats, in the event where a secret has been setup manually prior to be serviced via this cmdlet.

I think the options are to either include more the above as a comment in the code, or switch over to using the new property name for any new secrets being created; but I'm not inclined to migrate existing secrets with the old name, but feel free to persuade me otherwise 😀

MikeEvansLarah replied:
No I don't think migrating the existing secrets is necessary. Maybe just a code comment then with your commentary from above.

JamesDawson replied:
Having just seen a workflow failure, I see now that the old format does not work with at least the later version of the `azure/login` action we use.  So I've updated the implementation to catch when existing secrets are using the old format and force a rotate, similarly to the previous approach that only looked for the `password` property.

JamesDawson committed:
Add comment to provide additional context for the change and what hasn't been changed.

JamesDawson committed:
Simplified implementation to always apply the current KV property convention and updated tests.

Counter to prior understanding, the old format of key vault secret does not work with the current version of 'azure/login' GHA.

JamesDawson committed:
Update log message to fix strict mode test failure

JamesDawson committed:
Add support for storing 'subscriptionId' in the key vault secret, as this is a required property when using with the 'azure/login' GH Action.
